server/mdm/apple/profile_verifier.go

```diff
@@ -6,11 +6,61 @@ import (
+// protocol. The Fleet server periodically runs a cron that enqueues install profile
+// commands for host profiles that do not have a verification status (i.e. status is null).
+// Install profile commands may be enqueued as a result of a variety of events, such as when a host
+// enrolls in Fleet, when a host's team membership changes, when a new profile is uploaded, when an
+// existing profile is modified, or when a failed profile is retried.
+//
+// Verification status of a host profile can change in the following ways:
+//
+// - When an install profile command is enqueued by the server, the verification status is set to "pending".
+//
+// - When the results of an install profile command are reported via the MDM protocol, the Fleet server
+//   parses the results and updates the host's verification status for the applicable profile. If the
+//   command was acknowledged, the verification status is set to "verifying". If the command resulted
+//   in an error, the server determines if the profile should be retried (in which case, a new install profile
+//   command will be enqueued by the server) or marked as "failed" and updates the datastore accordingly.
+//
+// - When host details are reported via osqueery, the Fleet server ingests a list of installed
+//   profiles and compares the reported profiles with the list of profiles expected to be
+//   installed on the host. Expected profiles comprise the profiles that belong to the host's assigned
+//   team (or no  team, as applicable). If an expected profile is found, the verification status is
+//   set to "verified". If an expected profile is missing from the reported results, the server determines
+//   if the profile should be retried (in which case, a new install profile command will be enqueued by the server)
+//   or marked as "failed" and updates the datastore accordingly.
+
+// maxRetries is the maximum times an install profile command may be retried, after which marked as failed and no further
+// attempts will be made to install the profile.
+const maxRetries = 1
+
 // ProfileVerificationStore is the minimal interface required to get and update the verification
 // status of a host's MDM profiles. The Fleet Datastore satisfies this interface.
 type ProfileVerificationStore interface {
+	// GetHostMDMProfilesExpectedForVerification returns the expected MDM profiles for a given host. The map is
+	// keyed by the profile identifier.
 	GetHostMDMProfilesExpectedForVerification(ctx context.Context, host *fleet.Host) (map[string]*fleet.ExpectedMDMProfile, error)
-	UpdateHostMDMProfilesVerification(ctx context.Context, host *fleet.Host, verified, failed []string) error
+	// GetHostMDMProfilesRetryCounts returns the retry counts for the specified host.
+	GetHostMDMProfilesRetryCounts(ctx context.Context, hostUUID string) ([]fleet.HostMDMProfileRetryCount, error)
+	// GetHostMDMProfileRetryCountByCommandUUID returns the retry count for the specified
+	// command UUID and host UUID.
```

Comment:
Again, nit but just to avoid confusion in the order of args:
```suggestion
	// host UUID and command UUID.
```